OpenChange/MAPIStoreTasksFolder.m

```diff
@@ -116,18 +133,64 @@ - (uint32_t) exchangeRightsForRoles: (NSArray *) roles
            && [roles containsObject: SOGoCalendarRole_PrivateViewer]
            && [roles containsObject: SOGoCalendarRole_ConfidentialViewer])
     rights |= RightsReadItems;
+
+  if ([roles containsObject: MAPIStoreRightEditOwn])
+    rights |= RightsEditOwn;
+  if ([roles containsObject: MAPIStoreRightDeleteOwn])
+    rights |= RightsDeleteOwn;
+
+  if ([roles containsObject: MAPIStoreRightFolderOwner])
+    rights |= RightsFolderOwner | RoleNone;
+
+  if ([roles containsObject: MAPIStoreRightFolderContact])
+    rights |= RightsFolderContact;
+
   if (rights != 0)
     rights |= RoleNone; /* actually "folder visible" */
 
   return rights;
 }
 
+- (BOOL) subscriberCanModifyMessages
+{
+  static NSArray *modifierRoles = nil;
+
+  if (!modifierRoles)
+    modifierRoles = [[NSArray alloc] initWithObjects:
+                                       SOGoCalendarRole_PublicModifier,
```

Comment:
2 spaces :o, but I don't mind (i just want to put something because i can't spot any bug)